wallet/src/repository/wallet/mod.rs

```diff
@@ -59,14 +58,36 @@ struct AccountMutation {
 pub struct WalletUtxos<'a> {
     pub utxo_set: &'a model::UtxoSet,
     pub used_outputs: &'a mut model::UsedOutputs,
+    pub unconfirmed_transactions: &'a HashSet<Hash>,
 }
 
 impl<'a> OutputsCollection for WalletUtxos<'a> {
     fn sort_by(&self, strategy: UtxoSelectionStrategy) -> Vec<OutputPointer> {
+        let filter_unconfirmed = |out_ptr: &model::OutPtr| {
+            let pointer: OutputPointer = out_ptr.into();
+            if self
+                .unconfirmed_transactions
+                .contains(&pointer.transaction_id)
+            {
+                None
+            } else {
+                Some(pointer)
+            }
+        };
```

Comment:
We are still not sure if this if something a user would like or not... :smile: 

- This feature will make that transactions that require some unconfirmed UTXO, will have to wait at least 20 blocks.
- Some users might NOT want to able to create transactions from unconfirmed UTXOs (e.g. trades)
- Other users might prefer to have the freedom to create several transaction without having to wait for confirmation.